cli/azd/pkg/tools/swa.go

```diff
@@ -74,8 +76,33 @@ func (cli *swaCli) Deploy(ctx context.Context, tenantId string, subscriptionId s
 	return res.Stdout, nil
 }
 
+// base version number and empty string if there's no pre-request check on version number
+func (cli *swaCli) GetToolUpdate() ToolMetaData {
+	return ToolMetaData{
+		MinimumVersion: semver.Version{
+			Major: 1,
+			Minor: 0,
+			Patch: 0,
+		},
+		UpdateCommand: "Visit https://github.com/Azure/static-web-apps-cli/releases to install newer",
+	}
+}
+
 func (cli *swaCli) CheckInstalled(_ context.Context) (bool, error) {
-	return toolInPath("npx")
+	found, err := toolInPath("npx")
+	if !found {
+		return false, err
+	}
+	swaRes, _ := exec.Command("npx", "@azure/static-web-apps-cli", "--version").Output()
```

Comment:
I wonder if we even need to do the version checking here at all.  Perhaps instead we should just pass the version of the `static-web-apps-cli` we want to `npx` when we launch it (e.g. just change `defaultArgs` on line 117 to be `@azure/static-web-apps-cli@1.0.0` and just used a fixed version?

In that case, I think the check we'd want to do here is the version of `npx` (and the minimum version should be the same as the minimum version we have for `npm` since `npx` is just a wrapper over `npm` for running packages you may not have installed).

/cc @wbreza for his feedback on this.